docs/guides/tutorials/chatgpt_bot.rst

```diff
@@ -0,0 +1,2285 @@
+        apiKey: process.env.OPENAI_API_KEY!,
+      });
+    }
+
+It's pretty simple. It makes sure the API key was provided in the environment
+variable and returns a new API client initialized with that key.
+
+Now, let's create error messages we will use in a couple of places if these API
+calls go wrong. Create a file ``app/constants.ts`` and fill it with this:
+
+.. code-block:: typescript
+
+    export const errors = {
+      flagged: `OpenAI has declined to answer your question due to their
+              [usage-policies](https://openai.com/policies/usage-policies). Please try
+              another question.`,
+      default: "There was an error processing your request. Please try again.",
+    };
+
+This exports an object ``errors`` with a couple of error messages.
+
+Now, let's get the documentation ready!
+
+
+Put the documentation in place
+==============================
+
+For this project, we will be using documentation written as Markdown files
+since they are straightforward for OpenAI's language models to use.
+
+Create a ``docs`` folder in the root of the project. Here we will place our
+Markdown documentation files. You can grab the files we use from `the example
+project's GitHub repo
+<https://github.com/edgedb/edgedb-examples/tree/main/docs-chatbot/docs>`_ or
+add your own. (If you use your own, you may also want to adjust the system
+message we send to OpenAI later.)
+
+.. note:: On using formats other than Markdown
+
+    We *could* opt to use other simple formats like plain text files or more
+    complex ones like HTML. Since more complex formats can include additional
+    data beyond what we want the language model to consume (like HTML's tags
+    and their attributes), we may first want to clean those files and extract
+    the content before sending it to OpenAI. (We can write our own logic for
+    this or use libraries that are available online for conversion, to Markdown
+    for example.)
+
+    It's possible to use more complex formats *without* cleaning them, but then
+    we're paying for extra tokens that don't improve the answers our chatbot
+    will give users.
+
+.. note:: On longer documentation sections
+
+    In this tutorial project, our documentation pages are short, but in
+    practice, documentation files can get quite long and may need to be split
+    into multiple sections because of the LLM's token limit. LLMs divide text
+    into tokens. For English text, 1 token is approximately 4 characters or
+    0.75 words. LLMs have limits on the number of tokens they can receive and
+    send back.
+
+    One approach to mitigate this is to parse your documentation files and
+    create new sections every time you encounter a header. If you use this
+    approach, consider section lengths when writing your documentation. Use
+    your token limit to set a hard limit on documentation section length to
+    make sure it's accessible to your chatbot. If you find a section is too
```

Comment:
`Use your token limit to set a hard limit on documentation section length to make sure it’s accessible to your chatbot. ` I'm not sure about this sentence.

Maybe we can move this whole section to the end or just rephrase/remove the sentence from above: 
`One approach to mitigate this is to parse your documentation files and create new sections every time you encounter a header. If you use this approach, consider section lengths when writing your documentation. Use your token limit to set a hard limit on documentation section length to make sure it’s accessible to your chatbot. If you find a section is too long, consider ways you might break it up with additional headings. This will probably make it easier to read for your users too!`

Reply:
Yeah, I don't think it adds much. I'll remove it.